test/unit/alphabet/composite/alphabet_variant_test.cpp

```diff
@@ -332,3 +333,43 @@ TEST(alphabet_variant_test, two_different_variants)
     seqan3::alphabet_variant<seqan3::rna4, seqan3::gap> r2{'A'_rna4};
     EXPECT_EQ(l, r2); // this works because rna4 and dna4 are implicitly convertible to each other
 }
+
+TEST(alphabet_variant_test, char_is_valid_for)
+{
+    EXPECT_TRUE(seqan3::char_is_valid_for<seqan3::gapped<seqan3::rna5>>('A')); // valid seqan3::rna5 char
+    EXPECT_TRUE(seqan3::char_is_valid_for<seqan3::gapped<seqan3::rna5>>('a')); // valid seqan3::rna5 char
+    EXPECT_TRUE(seqan3::char_is_valid_for<seqan3::gapped<seqan3::rna5>>('-')); // valid seqan3::gap char
+    EXPECT_FALSE(seqan3::char_is_valid_for<seqan3::gapped<seqan3::rna5>>('S')); // neither seqan3::rna5 nor seqan3::gap
+
+    using char_t = seqan3::alphabet_char_t<seqan3::gapped<seqan3::rna5>>;
+
+    char_t i = std::numeric_limits<char_t>::min();
+    char_t end = std::numeric_limits<char_t>::max();
+
+    // note that without the cast i <= end would result in an ininite loop
+    for (; static_cast<int64>(i) <= static_cast<int64>(end); ++i)
+    {
+        EXPECT_EQ(seqan3::char_is_valid_for<seqan3::gapped<seqan3::rna5>>(i),
+                  seqan3::char_is_valid_for<seqan3::gap>(i) || seqan3::char_is_valid_for<seqan3::rna5>(i));
+    }
```

Comment:
```suggestion
    using gapped_alphabet_t = TypeParam;
    using gapped_alphabet_bases_t = typename gapped_alphabet_t::seqan3_required_types;
    using char_t = seqan3::alphabet_char_t<gapped_alphabet_t>;

    char_t i = std::numeric_limits<char_t>::min();
    char_t end = std::numeric_limits<char_t>::max();

    // note that without the cast i <= end would result in an ininite loop
    for (; static_cast<int64>(i) <= static_cast<int64>(end); ++i)
    {
        // is thar char valid for any of the bases?
        bool is_valid{};
        seqan3::detail::for_each<gapped_alphabet_bases_t>([&is_valid, i](auto id)
        {
             using type = typename decltype(id)::type;
             is_valid = is_valid || seqan3::char_is_valid_for<type>(i);
        });

        EXPECT_EQ(seqan3::char_is_valid_for<gapped_alphabet_t>(i), is_valid);
    }
```

Shouldn't this work? And we can make the test case a typed test?

Reply:
I'll remind you of this conversation when I request a change and you insist that it is not the scope of the current PR